Source/WebKit/WPEPlatform/wpe/WPEMonitor.cpp

```diff
@@ -398,6 +440,64 @@ void wpe_monitor_set_size(WPEMonitor* monitor, int width, int height)
     }
 }
 
+/**
+ * wpe_monitor_get_available_width:
+ * @monitor: a #WPEMonitor
+ *
+ * Get the available width of @monitor in logical coordinates. This is the
+ * width of the monitor excluding any reserved areas like docks or panels.
+ *
+ * Returns: the available width of @monitor, or the full monitor width if available width could not be determined.
+ */
+int wpe_monitor_get_available_width(WPEMonitor* monitor)
+{
+    g_return_val_if_fail(WPE_IS_MONITOR(monitor), -1);
+
+    return monitor->priv->availableWidth != -1 ? monitor->priv->availableWidth : monitor->priv->width;
+}
+
+/**
+ * wpe_monitor_get_available_height:
+ * @monitor: a #WPEMonitor
+ *
+ * Get the available height of @monitor in logical coordinates. This is the
+ * height of the monitor excluding any reserved areas like docks or panels.
+ *
+ * Returns: the available height of @monitor, or the full monitor height if available height could not be determined.
+ */
+int wpe_monitor_get_available_height(WPEMonitor* monitor)
+{
+    g_return_val_if_fail(WPE_IS_MONITOR(monitor), -1);
+
+    return monitor->priv->availableHeight != -1 ? monitor->priv->availableHeight : monitor->priv->height;
+}
+
+/**
+ * wpe_monitor_set_available_size:
+ * @monitor: a #WPEMonitor
+ * @width: the available width, or -1
+ * @height: the available height, o -1
+ *
+ * Set the available size of @monitor in logical coordinates. This is the
+ * size of the monitor excluding any reserved areas like docks or panels.
+ */
+void wpe_monitor_set_available_size(WPEMonitor* monitor, int width, int height)
+{
+    g_return_if_fail(WPE_IS_MONITOR(monitor));
+    g_return_if_fail(width == -1 || width >= 0);
+    g_return_if_fail(height == -1 || height >= 0);
+
```

Comment:
Add g_object_freeze_notify/g_object_thaw_notify to make sure when notify is emitted for eny of the properties, botah have been updated